src/main/java/org/mastodon/revised/util/GroupStrings.java

```diff
@@ -0,0 +1,252 @@
+package org.mastodon.revised.util;
+
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Deque;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Utility class used to group a collection of strings by their common first
+ * words.
+ * <p>
+ * For instance, the 3 strings:
+ * 
+ * <pre>
+ * - "Save Project"
+ * - "Import TGMM tracks"
+ * - "Import Simi BioCell tracks"
+ * </pre>
+ *
+ * will be grouped in 2 groups:
+ * 
+ * <pre>
+ * - "Save Project"
+ * - "Import"
+ * 	+ "TGMM tracks"
+ * 	+ "Simi BioCell tracks"
+ * </pre>
+ * 
+ * Comparisons of common words are case insensitive.
+ * 
+ * @author Jean-Yves Tinevez
+ */
+public class GroupStrings
+{
+
+	private final List< String > names;
+
+	public GroupStrings()
+	{
+		this.names = new ArrayList<>();
+	}
+
+	public void add( final String string )
+	{
+		names.add( string );
+	}
+
+	public Collection< Group > group()
+	{
+		final List< Group > sgs = new ArrayList<>();
+		for ( final String name : names )
+			sgs.add( new Group( name ) );
+
+		/*
+		 * Fuse.
+		 */
+		final List< Group > fused = fuseCollection( sgs );
+
+		/*
+		 * Split.
+		 */
+		final List< Group > split = new ArrayList<>();
+		for ( final Group sg : fused )
+			split.addAll( sg.split() );
+		
+		return split;
+	}
+
+	private static List< Group > fuseCollection( final Collection< Group > c )
+	{
+		final Deque< Group > champions = new ArrayDeque<>( c );
+		final List< Group > fused = new ArrayList<>();
+		if ( champions.size() > 1 )
```

Comment:
If input contains 1 group, `fuseCollection` will return an empty list. Is this a bug?